app/pages/preview_post.tsx

```diff
@@ -58,20 +59,31 @@ export const getServerSideProps: GetServerSideProps<PageProps> = async ({
 
 export default function Preview({ configuratorState, locale }: PageProps) {
   i18n.activate(locale);
-  const [migrated] = useState(() =>
-    migrateConfiguratorState({ ...configuratorState, state: "PUBLISHED" })
-  );
+  const [migrated, setMigrated] = useState<ConfiguratorState>();
+  useEffect(() => {
+    const run = async () => {
+      setMigrated({
+        ...(await migrateConfiguratorState(configuratorState)),
+        state: "PUBLISHED",
+      } as ConfiguratorState);
+    };
+
+    run();
+  }, [configuratorState]);
```

Comment:
Nit: Maybe we should have a useMigratedConfiguratorState hook here. It seems that this pattern will appear multiples (already is there from the homepage), and it seems we could be in need of having an object like { data, isLoading } similar to react useQuery. Having a central place to put this pattern could be good.

Reply:
I checked the code and looks like in the end this is the only one place that uses this in a React-component context (except of Examples that use a bit different pattern in a component context), all the other ones are inside regular functions, maybe it's fine to not introduce a new concept just for this one page?